pyanaconda/modules/storage/bootloader/installation.py

```diff
@@ -108,16 +110,22 @@ def run(self):
 class InstallBootloaderTask(Task):
     """Installation task for the bootloader."""
 
-    def __init__(self, storage, mode):
+    def __init__(self, storage, mode, payload_type, sysroot):
         """Create a new task."""
         super().__init__()
         self._storage = storage
         self._mode = mode
+        self._payload_type = payload_type
+        self._sysroot = sysroot
 
     @property
     def name(self):
         return "Install the bootloader"
 
+    def _have_bootupd(self):
+        """Is bootupd/bootupctl present in sysroot?"""
+        return os.path.exists(join_paths(self._sysroot, "/usr/bin/bootupctl"))
```

Comment:
I'm not happy about this misplaced condition. I mean if we change the logic in RPM_OSTREE installation part it won't be reflected here. Ideally we should have it decided by rpm_ostree installation and just pass the decision here.

However, I think I see the reasoning so I'm fine with that.

Reply:
Perhaps a single source of the condition would be good to have?